### DISCUSSION TITLE
HTML output: is there a way to style only one field of a pandas dataframe?

### DISCUSSION THREAD

**Discussion by aborruso** [quarto-dev/quarto-cli#4073] [Q&A] [answered]
Hi,
if I use `set_table_styles`

~~~~
```{python}
#| echo: false
#| warning: false

import pandas as pd

df = pd.DataFrame({'text': ['foo', 'bar'],'number': [1, 2]})

dfStyler = df.style.set_properties(**{'text-align': 'center'})
dfStyler.set_table_styles([dict(selector='th', props=[('text-align', 'center')])])
```
~~~~

I have in the notebook the style applied

![image](https://user-images.githubusercontent.com/30607/213640152-cd282f9e-d7cd-4cbd-aa52-6b40188ee199.png)

If I run it in quarto, styling is managed by bootstrap, and I have the default left align

![image](https://user-images.githubusercontent.com/30607/213640307-f60b92c6-ad34-4028-b884-7dfb1f99dc33.png)

But I can apply some style in this way:

```
df.style.hide_index().set_table_attributes("class='dataframe table-hover text-center'")
```

to have 

![image](https://user-images.githubusercontent.com/30607/213642910-9d2c13da-8aa2-4b94-87c9-d960e4ba8c27.png)

But how to apply in example right align only for the `number` column?

If I set it in this way:

```
df.style.set_properties(subset=["number"], **{'text-align': 'right'})
```

it works in the notebook, but quarto style forces `number` column to align to the left

![image](https://user-images.githubusercontent.com/30607/213643738-7eff44e2-e4d6-4b1c-b7f7-145a57438c8b.png)

Is there a way to style only one field of a pandas dataframe?

Thank you

**Comment by cscheid** [comment]
You'll probably have better luck asking this in a pandas forum, since it's really a pandas question.

**Reply by aborruso** [reply]
> You'll probably have better luck asking this in a pandas forum, since it's really a pandas question.

I will try to do it, but I think I won't be able to write it right, because I'm not a pandas user and I don't know how to replicate the construction of an output non-aligned with the input.

I think it's really a particularly important feature for quarto, since it's a really great open-source scientific and technical publishing system.

**Reply by aborruso** [reply]
I wrote this: https://groups.google.com/g/pandoc-discuss/c/RigoN8s9sH0

Thank you @cscheid

**Reply by cscheid** [reply]
You seem to have asked a question on `pandoc-discuss`. I suggested you ask on the *`pandas`* discussion forum, since you are using the data frame library Pandas. I think you confused "Pandas" and Pandoc.

> I think it's really a particularly important feature for quarto, since it's a really great open-source scientific and technical publishing system.

To be clear: we do not control the API that Pandas uses to generate tables. They wrote the code.

**Reply by aborruso** [reply]
> I think you confused "Pandas" and Pandoc.

It's a difficult time for me and I do stupid things. I have to slow down.
Thank you

**Reply by aborruso** [reply]
> You'll probably have better luck asking this in a pandas forum, since it's really a pandas question.

Hi @cscheid forgive me for coming back here, but there is something I don't understand in what you write.
Why do you say it's a pandas problem?

In pandas if I set this 

```
import pandas as pd

df = pd.DataFrame({'text': ['foo', 'bar'],'number': [1, 2]})

df.style.set_properties(subset=["number"], **{'text-align': 'right'})
```

I'm able to format only one column and align it to the right, in my jupyter notebook

![image](https://user-images.githubusercontent.com/30607/214437096-275d1829-601c-45d7-b234-d9da67698800.png)

If I render it in HTML via quarto I have all aligned to the left.
Then it seems to me a quarto question.

Also @AlFontal writes about these problems in #3964

![image](https://user-images.githubusercontent.com/30607/214437473-4ae41ed4-cf5e-4c6e-97af-c615c85168b1.png)

**Reply by cscheid** [reply]
... That's because I misread your message 🤦 I'm so sorry.

It wasn't clear at first reading that you had output _which works on jupyter but not on quarto_. Let me investigate right away.

**Reply by cscheid** [reply]
Fix incoming: https://github.com/quarto-dev/quarto-cli/pull/4088

**Reply by aborruso** [reply]
Wow this will be a great fix!!

**Comment by aborruso** [comment, marked as answer]
The one of @cscheid is the reply: https://github.com/quarto-dev/quarto-cli/discussions/4073#discussioncomment-4771968
